### PR TITLE
SCC-1420

### PR DESCRIPTION
**Pull request by owen-opencast**

### CODE REVIEW

File: app/config/FrontendAppConfig.scala [hunk 1]
```diff
@@ -123,6 +123,12 @@ class FrontendAppConfig @Inject() (configuration: Configuration) {
   val preferencesFrontendTimeout: Int = configuration.get[Int]("external-url.preferences-frontend.timeoutInSec")
   val agentAuthorisationTimeout: Int  =
     configuration.get[Int]("microservice.services.agent-client-authorisation.timeoutInSec")
+  val addressLookupTimeout: Int       =
+    configuration.get[Int]("microservice.services.address-lookup.timeoutInSec")
```
Comment (owen-opencast) [inline app/config/FrontendAppConfig.scala:127]: timeouts for each of the connectors

File: app/connectors/CitizenDetailsConnector.scala [hunk 2]
```diff
@@ -16,65 +16,116 @@
...
+import uk.gov.hmrc.http.client.HttpClientV2
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse, InternalServerException, JsValidationException, NotFoundException, StringContextOps, Upstream4xxResponse, Upstream5xxResponse, UpstreamErrorResponse}
 import utils.HeaderUtils
 
 import javax.inject.Inject
+import scala.concurrent.duration.{Duration, SECONDS}
 import scala.concurrent.{ExecutionContext, Future}
 
-class CitizenDetailsConnector @Inject() (http: HttpClient, appConfig: FrontendAppConfig)(implicit
+class CitizenDetailsConnector @Inject() (http: HttpClientV2, appConfig: FrontendAppConfig)(implicit
```
Comment (owen-opencast) [inline app/connectors/CitizenDetailsConnector.scala:35]: Upgraded all HttpClients to v2

File: app/controllers/changeaddress/CheckYourAnswersController.scala [hunk 3]
```diff
@@ -88,48 +89,64 @@ class CheckYourAnswersController @Inject() (
...
+              addressType = addressType,
+              startDate = startDate,
+              endDate = request.userAnswers
+                .get(RemovePostalAddressPage)
+                .filter(_ == Main)
+                .map(_ => LocalDate.now)
+            )
+            .map {
+              case CreateAddressSuccess                     =>
+                handleSuccess(oldAddress, newAddressWithStartDate, addressType)
```
Comment (owen-opencast) [inline app/controllers/changeaddress/CheckYourAnswersController.scala:105]: Address locking now happens only on success

File: app/controllers/changeaddress/WhatIsYourNewAddressUkController.scala [hunk 4]
```diff
@@ -88,7 +88,7 @@ class WhatIsYourNewAddressUkController @Inject() (
               if (searchResults.isEmpty) {
                 val formWithErrors =
                   form.bindFromRequest().copy(errors = Seq(FormError("postcode", "address.postcode.error.notFound")))
-                BadRequest(view(formWithErrors, mode, addressType, request.trustedHelper))
+                NotFound(view(formWithErrors, mode, addressType, request.trustedHelper))
```
Comment (owen-opencast) [inline app/controllers/changeaddress/WhatIsYourNewAddressUkController.scala:91]: Changed reporting for no results to match PTA

File: app/viewmodels/checkAnswers/ContactPreferencesSummary.scala [hunk 5]
```diff
@@ -27,31 +27,29 @@ import viewmodels.implicits._
...
-              ).filter(_ => request.trustedHelper.isEmpty)
-            ).withCssClass("contact-preferences")
-          )
-        }
-        .getOrElse(Seq.empty)
-
-    }
-
+    for {
+      maybeStatus <- preferencesService.paperlessStatus
```
Comment (owen-opencast) [inline app/viewmodels/checkAnswers/ContactPreferencesSummary.scala:37]: Moved this to service with error checking to fix crashes on timeout

**Review by ccatosdevelopment — APPROVED** [review]

**Pull request merged**